### ISSUE TITLE
Can't download source image from S3

### ISSUE THREAD

**Issue by scrazy77** [imgproxy/imgproxy#1368]
In v3.23 and early version the S3 download success 
log:
```
INFO    [2024-11-07T13:53:54Z] Started /insecure/rs:fit:600:600/plain/s3://bucket/path/%25E6%259B%25B8_114.jpeg  request_id=NsymNJB2qp_Ivv-z-wDj1 method=GET client_ip=10.0.181.16
INFO    [2024-11-07T13:53:54Z] Completed in 201.934241ms /insecure/rs:fit:600:600/plain/s3://bucket/path/%25E6%259B%25B8_114.jpeg  request_id=NsymNJB2qp_Ivv-z-wDj1 method=GET status=200 processing_options="Width: 600; Height: 600; Format: jpeg" client_ip=10.0.181.16 image_url="s3://bucket/path/%E6%9B%B8_114.jpeg"`
```
 v3.25 ~ v3.26.1 the S3 download failed
log:

```
INFO    [2024-11-07T13:55:33Z] Started /insecure/rs:fit:600:600/plain/s3://bucket/path/%25E6%259B%25B8_114.jpeg  request_id=KJslFXrkQuE3-OBNOH4Ln method=GET client_ip=10.0.181.16
ERROR   [2024-11-07T13:55:33Z] Completed in 150.99321ms /insecure/rs:fit:600:600/plain/s3://bucket/path/%25E6%259B%25B8_114.jpeg  request_id=KJslFXrkQuE3-OBNOH4Ln method=GET status=404 client_ip=10.0.181.16 error="Can't download source image: Status: 404; operation error S3: GetObject, https response error StatusCode: 404, RequestID: WXAXVXMA05S09D3V, HostID: 0l9/STnFC5RrmaOOMbQBC8XiWtG+dHEpvUpAxeOEopID8wa8x9k13Y7fMhizfBCMQEsrAjpjkEre6jG2K0zOwSInNtsXWaz8, NoSuchKey: "
```
Maybe the filename with '%25' cause this 404 ...

**Comment by DarthSim** [comment]
Hey @scrazy77?

What's the real key of the object in S3? Does it work if you use `s3://bucket/path/%E6%9B%B8_114.jpeg` source URL?

**Comment by scrazy77** [comment]
s3://bucket/path/%E6%9B%B8_114.jpeg works! but don't know why different results between 2 versions.

**Comment by DarthSim** [comment]
That's because we fixed the handling of escaped symbols in non-HTTP(S) source URLs in `v3.25.0`.

I guess the filename in S3 is `書_114.jpeg`. `%25E6%259B%25B8_114.jpeg` is a twice-URL-encoded filename. Double encoding is a workaround for the bug fixed in `v3.25.0`.
